genapi/src/elem_type.rs

```diff
@@ -310,12 +310,10 @@ impl RegPIndex {
         store: &impl NodeStore,
         cx: &mut ValueCtxt<T, U>,
     ) -> GenApiResult<i64> {
-        let base = self
-            .p_index
-            .expect_iinteger_kind(store)?
-            .value(device, store, cx)?;
+        let base = self.p_index.value(device, store, cx)?;
```

Comment:
I felt it'd be a little dangerous to set an arbitrary value to an enum node. But, since `IValue` is just used internally to cast a type by following the given GenICam XML, I changed it to allow setting an arbitrary value to an enum node.